src/include/sof/dai.h

```diff
@@ -169,6 +169,12 @@ void dai_put(struct dai *dai);
 #define dai_fifo(dai, direction) \
 	dai->plat_data.fifo[direction].offset
 
+#if defined(CONFIG_CANNONLAKE) || defined(CONFIG_ICELAKE) \
+	|| defined(CONFIG_SUECREEK)
+/* DMIC power ON bit for I */
+#define DMIC_POWER_ON	((uint32_t) BIT(0))
```

Comment:
Use register bit name instead of DMIC_POWER_ON

Reply:
Starting the clock and power simultaneously caused the microphone to capture something that looked like square wave (test condition that @mrajwa uses is loud sine wave). The microphones in the device have a different acoustical to digital dBFS/dBSPL sensitivity for sensor mode and normal mode. The square wave like audio waveform would be explained be erroneously entering the sensor mode. Since this is more complicated than what I expected I agreed that moving power on to earlier is OK despite the small power hit.

We need to study what is the minimum time between the events for correct normal mode start and consider splitting the start sequence and implement such later if there is a significant improvement seen for power. If suspend is used in this platform it might provide already most of the power save opportunity. Just need to ensure the power on and clock start events have the minimum separation needed due to the various system resume delays to not get accidentally what we saw yesterday.

Reply:
DMIC_POWER_ON is just a single bit not a register. According to documentation "Set Power Active (SPA)" bit. I think DMIC_POWER_ON is better than a bare 0x01 or SPA as documentation calls it.